src/pages/create_blog.jsx

```diff
@@ -0,0 +1,158 @@
+import { useState } from "react";
+import { collection, addDoc } from "@firebase/firestore";
+import { db, storage } from "@/firebase/config";
+import { ref, uploadBytes } from "firebase/storage";
+import Button from "@/components/ui/Button";
+import PageIntro from "@/components/PageIntro";
+import { withTranslation } from "next-i18next";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { useRouter } from "next/navigation";
+
+export const extractFirstName = (str) => {
+    const firstSpaceIndex = str.indexOf(" ");
+    if (firstSpaceIndex === -1) {
+        return str;
+    }
+    return str.slice(0, firstSpaceIndex).toLowerCase();
+};
+const CreateBlog = ({ t }) => {
+    const [enTitle, SetEnTitle] = useState("");
+    const [arTitle, setArTitle] = useState("");
+    const [enPostText, setEnPostText] = useState("");
+    const [arpostText, setArPostText] = useState("");
+    const [imageUpload, setImageUpload] = useState(null);
+    const postCollection = collection(db, "blogs");
+    const router = useRouter();
+    const uploadImage = () => {
+        if (imageUpload == null) return;
+        const imageRef = ref(
+            storage,
+            `blogImages/${extractFirstName(enTitle)}.jpeg`
+        );
+        uploadBytes(imageRef, imageUpload).then(() => alert("Image uploaded"));
+    };
+
+    const setData = async () => {
+        try {
+            await uploadImage();
```

Comment:
if you can make it reusable inside firebase folder as others functions

Reply:
I just make a 2 functions to (upload Image,Download Image)